pkg/kubelet/kubelet_pods.go

```diff
@@ -1114,11 +1114,11 @@ func (kl *Kubelet) HandlePodCleanups() error {
 	}
 	for _, runningPod := range runningRuntimePods {
 		switch workerState, ok := workingPods[runningPod.ID]; {
-		case ok && workerState == SyncPod, ok && workerState == TerminatingPod:
+		case ok && workerState == SyncPod, ok && workerState == TerminatingPod, ok && workerState == TerminatedPod:
```

Comment:
I thank this change could not fix your problem. If a pod is already in terminated and it's podworker has just been removed and the housekeeping loop sees an
out of date cache entry for a running container,(https://github.com/kubernetes/kubernetes/blob/64e83a7e43afd038b3d5a0179e85099a0958de4c/pkg/kubelet/kubelet_pods.go#L1107), then the podworker would still keep killing a NotFound pod and being stucked until next kubelet restart.

Reply:
Thanks for reminding. Yes, if `HandlePodCleanups` runs another time after the Pod is removed from `podSyncStatuses` and the Pod is found in `runtimeCache`, it would still call UpdatePod. I guess the change fixed my problem because the cache's cachePeriod is 2 seconds, which is same as housekeepingPeriod. Then next housekeeping loop won't see the outdated Pod (since all of its containers and sandboxes have been dead 2s ago).
However, it's obscure to rely on the periods and could be easily broken in future. I have changed to use another way to avoid creating an active podSyncStatus when the runtime Pod is already terminated, similar to https://github.com/kubernetes/kubernetes/pull/105527. PTAL.